src/test/java/org/apache/commons/compress/compressors/gzip/GzipParametersTest.java

```diff
@@ -46,4 +50,44 @@ public void testToString() {
         gzipParameters.setOS(GzipParameters.OS.Z_SYSTEM);
         assertTrue(gzipParameters.toString().contains("Z_SYSTEM"));
     }
+
+    @ParameterizedTest
+    //@formatter:off
+    @CsvSource({
+        "          , helloworld",
+        "          , helloéworld",
+        "ISO-8859-1, helloworld",
+        "ISO-8859-1, helloéworld",
+        "UTF-8     , helloworld",
+        "UTF-8     , helloéworld"
+    })
+    //@formatter:on
+    public void testLegalCommentOrFileName(final String charset, final String text) {
+        final GzipParameters p = new GzipParameters();
+        if (charset != null) {
+            p.setFileNameCharset(Charset.forName(charset));
```

Comment:
@ddeschenes-1 
In the future, make the parameter a `Charset` instead of a String and you won't need the test to do the conversion.

Reply:
There is no automatic converter from String to Charset in junit5.
Adding the necessary lines for a @ConvertWith converter class would be far more code than those 2 lines.

Reply:
Except when you're wrong: https://github.com/junit-team/junit5/blob/main/junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/StringToCommonJavaTypesConverter.java#L38

Reply:
Great. thank you internet for misleading me again. what is it gonna be in 10 more years...

btw, when is 1.28.0 releasing?

Reply:
I'd first like to finish shepherding the current other Apache Commons release candidates to completion. Could be another week or two before this components gets a release candidate.